### PR TITLE
Feat: Use Security Group Module; Restrict MSK Ingress Based on Enabled Protocols

### LINKED ISSUE #5: input variable `security_groups` is not being utilized
Kafka brokers are only using the default security groups.  Passing in variable security_groups is not being used.

main.tf
```hcl
resource "aws_msk_cluster" "default" {
  count                  = module.this.enabled ? 1 : 0
  cluster_name           = module.this.id
  kafka_version          = var.kafka_version
  number_of_broker_nodes = var.number_of_broker_nodes
  enhanced_monitoring    = var.enhanced_monitoring

  broker_node_group_info {
    instance_type   = var.broker_instance_type
    ebs_volume_size = var.broker_volume_size
    client_subnets  = var.subnet_ids
    security_groups = aws_security_group.default.*.id
  }
}
```

Line `security_groups = aws_security_group.default.*.id` does not add variable `security_groups`.  Maybe replace with `security_groups = concat(aws_security_group.default.*.id, var.security_groups)`

### LINKED ISSUE #16: Restrict security group to tls port 9094
Would be good if security group created open ports related to kafka 9094 (tls)

### PR DESCRIPTION
**Pull request by korenyoni**
## what
* Use standardized Cloud Posse Security Group convention (https://github.com/cloudposse/terraform-aws-security-group)
* Restrict Security Group Ingress based on enabled MSK protocols.
* Bump minimum Terraform version to support SG module.

## why
* The new Security Group standards include an input `var.associated_security_group_ids` which allows additional security groups to be associated with the MSK brokers, removing the need for `var.broker_node_security_groups`, hence why #41 is superseded by this PR.
* The Security Groups needs to be restricted based on which protocols are enabled.
* Abstract unnecessary SG-related logic (especially present when dynamically enabling or disabling various protocols for both SG-based and CIDR-based SG rules) via SG module.

## references
* co-authored by @nitrocode 
* Supersedes and closes #41 
* Closes #5
* Closes #16 
* https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/msk_cluster#security_groups
* https://docs.aws.amazon.com/msk/latest/developerguide/client-access.html#port-info
* Similar PR: https://github.com/cloudposse/terraform-aws-elasticache-redis/pull/133

### CODE REVIEW

**Review by Nuru — DISMISSED** [review]

File: security_group_inputs.tf [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+  description = <<-EOT
+    The description to assign to the created Security Group.
+    Warning: Changing the description causes the security group to be replaced.
+    EOT
+}
+
+variable "security_group_create_before_destroy" {
+  type = bool
+
+  default     = false
```
[the remark this answers is not in the document]
Comment (korenyoni) [reply security_group_inputs.tf]: @Nuru I was basing this on the instructions here:

https://github.com/cloudposse/terraform-aws-security-group/blob/c6e4156696ee28cad0cd927c82377fbd73156199/exports/security_group_inputs.tf#L71-L72

> If the resource has to be deleted to change its security group,
> then set the default to `false` and highlight the option to change
> it to `true` in the release notes and migration documents.

File: main.tf [hunk 2]
```diff
@@ -12,47 +12,86 @@ locals {
...
+  source  = "cloudposse/security-group/aws"
+  version = "0.4.2"
+
+  enabled                       = local.enabled && var.create_security_group
+  security_group_name           = var.security_group_name
+  create_before_destroy         = var.security_group_create_before_destroy
+  security_group_create_timeout = var.security_group_create_timeout
+  security_group_delete_timeout = var.security_group_delete_timeout
+
+  security_group_description = coalesce(var.security_group_description, "Allow inbound traffic from Security Groups and CIDRs. Allow all outbound traffic")
```
[the remark this answers is not in the document]
Comment (korenyoni) [reply main.tf]: resolved

**Review by Nuru — APPROVED** [review]

**Pull request merged**